### PR TITLE
✨ File Magic Detection & NGFF SQLite3 Support

### PR DESCRIPTION
**Pull request by John-P**
Adds a module for detection of file types via magic number/signatures for:
- zip via `is_zip`
- sqlite3 via `is_sqlite`
- DICOM (.dcm) via `is_dcm`

This PR also allows for loading of a zarr (NGFF) from an SQLite3 database (because of `is_sqlite`).

### CODE REVIEW

File: tiatoolbox/utils/magic.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        file (Union[str, Path, bytes]):
+            The file to check.
+
+    """
+    header = _normalize_file(file, 16)
+
+    return header == b"SQLite format 3\x00"
+
+
+def is_zip(file: Union[str, Path, bytes]) -> bool:
```
[the remark this answers is not in the document]
Comment (John-P) [reply tiatoolbox/utils/magic.py:41]: Yeah, can use zip file. It's a more robust check. I was implementing a simple bytes check to make the code pattern across lots of different magic number checks but happy to also use a library call if it make sense and isn't slow.

[the remark this answers is not in the document]
Comment (John-P) [reply tiatoolbox/utils/magic.py]: Changed to use zipfile now.

File: tests/test_wsireader.py [hunk 2]
```diff
@@ -1929,6 +1931,21 @@ def test_is_ngff_regular_zarr(tmp_path):
         WSIReader.open(zarr_path)
 
 
+def test_is_ngff_sqlite3(tmp_path, remote_sample):
+    """Test is_ngff is false for a sqlite3 file.
+
+    Copies the ngff-1 sample to a sqlite3 file and checks that it is
+    identified as an ngff file.
+    """
```
Comment (shaneahmed) [inline tests/test_wsireader.py:1939]:
```suggestion
    
    """
```

File: tiatoolbox/utils/magic.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+"""Detection of file type via magic numbers / signatures.
+
+Checks here are based on the file signature, not the file extension.
+They are all intented to be fast and lightweight, and should not require
+parsing the entire file. There may occationally be false positives which
+should be caught when attemping to parse the file.
+"""
```
Comment (shaneahmed) [inline tiatoolbox/utils/magic.py:7]:
```suggestion

"""
```

**Review by shaneahmed — APPROVED** [review]

**Pull request merged**